plugins/UFM_NDT_Plugin/ufm_sim_web_service/topo_diff/topo_diff.py

```diff
@@ -283,6 +291,58 @@ def get_request(host_ip, protocol, resource, headers):
     logging.info("UFM API Request Status: {}, URL: {}".format(response.status_code, request))
     return response
 
+
+def save_cable_validation_cookies(requests_cookiejar, cookie_filename):
+    '''
+    Save cookies to file
+    :param requests_cookiejar:
+    :param cookie_filename:
+    '''
+    with open(cookie_filename, 'wb') as f:
+        pickle.dump(requests_cookiejar, f)
+
+def load_cable_validation_cookies(cookie_filename):
+    '''
+    Load cookie from file
+    :param cookie_filename:
+    '''
+    with open(cookie_filename, 'rb') as f:
+        return pickle.load(f)
+
+def post_request_with_cookies(host_addr, port_num, username, password):
+    '''
+    Create cookie file and then use it for get request
+    :param host_addr:
+    :param port_num:
+    :param username:
+    :param password:
+    :param url:
+    '''
+    send_url = f"{CABLE_VALIDATION_PROTOCOL}://{host_addr}:{port_num}/{CABLE_VALIDATION_LOGIN_URL}"
+    send_payload = {'httpd_username': username, 'httpd_password': password}
+    try:
+        session = requests.Session()
+        rest_respond = session.post(send_url,data=send_payload,verify=False,
+                                            timeout=REST_TIMEOUT)
+        if rest_respond.status_code != SUCCESS_CODE:
+            logging.error(f"Cable validation login request failed: {send_url} return code {rest_respond.status_code}")
+            return None
+        save_cable_validation_cookies(session.cookies, COOKIE_FILE_PATH)
+    except Exception as e:
+        logging.error(f"Cable validation login request failed: {send_url} error: {e}")
+        return None
+    # send request for cable validation report
+    cv_report_url = f"{CABLE_VALIDATION_PROTOCOL}://{host_addr}:{port_num}/{CABLE_VALIDATION_REPORT_URL}"
+    try:
+        session = requests.Session()
+        cv_rest_respond = session.get(cv_report_url,
+                        cookies=load_cable_validation_cookies(COOKIE_FILE_PATH),
+                        verify=False,
+                        timeout=REST_TIMEOUT)
+    except Exception as e:
+        print(e)
+    return cv_rest_respond
```

Comment:
what are you returning here in case the report request failed with exception?
is it handled on the caller side?

Reply:
fixed